torch/csrc/jit/runtime/register_prim_ops.cpp

```diff
@@ -2489,9 +2490,46 @@ static const std::vector<OperatorGeneratorArgs> opGenArgs1{
     OperatorGeneratorArgs(
         // TODO return generator object when torchscript supports RNG
         // first-class
-        TORCH_SELECTIVE_SCHEMA("aten::manual_seed(int seed) -> ()"),
+        TORCH_SELECTIVE_SCHEMA("aten::manual_seed.default_(int seed) -> ()"),
         [](Stack& stack) { at::manual_seed(pop(stack).toInt()); },
         aliasAnalysisFromSchema()),
+    OperatorGeneratorArgs(
+        TORCH_SELECTIVE_SCHEMA(
+            "aten::Generator(*, str? device=None, int? seed=None) -> Generator"),
+        [](Stack& stack) {
+          auto seed = pop(stack).toOptional<int64_t>();
+          auto device_str = pop(stack);
+          auto device = c10::Device(
+              device_str.isString() ? device_str.toStringRef() : "cpu");
+          push(stack, at::make_generator_for_device(device, seed));
+        },
+        aliasAnalysisFromSchema()),
+    OperatorGeneratorArgs(
+        TORCH_SELECTIVE_SCHEMA("aten::initial_seed(Generator self) -> int"),
+        [](Stack& stack) {
+          auto generator = pop(stack);
+          auto current_seed = generator.toGenerator().current_seed();
+          push(stack, (int64_t)current_seed);
+        },
+        aliasAnalysisFromSchema()),
+    OperatorGeneratorArgs(
+        TORCH_SELECTIVE_SCHEMA(
+            "aten::manual_seed.generator(Generator self, int seed) -> Generator"),
+        [](Stack& stack) {
+          auto seed = pop(stack).toInt();
+          auto generator = pop(stack);
+          generator.toGenerator().set_current_seed(seed);
+          push(stack, generator);
+        },
+        aliasAnalysisFromSchema()),
```

Comment:
@eellison I also seem to now be failing the [backwards compatibility test suite](https://github.com/pytorch/pytorch/actions/runs/6500762113/job/17657211221?pr=110413). Its because I had to add an override for `aten::manual_seed` (as highlighted above). But because the previously existing `aten::manual_seed` didn't have any overrides it was failing the mobile test suites because it thinks it is registering `aten::manual_seed` multiple times. This is why I added the `aten::manual_seed.default_` override, but that is causing the backwards compatibility test to fail now. Can you please advise on how to proceed?

Reply:
is it possible to leave `aten::manual_seed` as is and then also add `aten::manual_seed.generator`? Not sure if it will work but I did see some other examples suggesting this might work - e.g. aten::device

Reply:
I tried this, but as I pointed out, it caused the following issue:

> it was failing the mobile test suites because it thinks it is registering aten::manual_seed multiple times. This is why I added the aten::manual_seed.default_ override